todo/src/components/TodoForm.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { addTodo, clearCompleted } from '../actions';
+import { Button } from 'reactstrap';
+import { InputGroup, InputGroupText, InputGroupAddon, Input } from 'reactstrap';
+
+class TodoForm extends React.Component {
+  state = {
+    newTodo: ''
+  };
+
+  handleChanges = e => {
+    this.setState({ newTodo: e.target.value });
+  };
+
+  addTodo = e => {
+    e.preventDefault();
+    this.props.addTodo(this.state.newTodo);
+    this.setState({ newTodo: '' });
+  };
+
+  clearCompleted = e => {
+    e.preventDefault();
+    this.props.clearCompleted();
+  };
+
+  render() {
+    return (
+      <form>
```

Comment:
```suggestion
      <form onSubmit={this.addTodo}>
```
instead of using it on the button to prevent a page refresh